perf/hackbench.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+# Based on code by Nikhil Rao <ncrao@google.com>
+#   copyright: 2008 Google
+#   https://github.com/autotest/autotest-client-tests/tree/master/hackbench
+
+import os
+import shutil
+import json
+
+from avocado import Test
+from avocado import main
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class Hackbench(Test):
+
+    """
+    This module will run the hackbench benchmark. Hackbench is a benchmark for
+    measuring the performance, overhead and scalability of the Linux scheduler.
+    The C program was pick from Ingo Molnar's page.
+    """
+
+    def setUp(self):
+        '''
+        Build Hackbench
+        Source:
+        http://people.redhat.com/~mingo/cfs-scheduler/tools/hackbench.c
+        '''
+        self.results = None
+        sm = SoftwareManager()
+        if not sm.check_installed("gcc") and not sm.install("gcc"):
+            self.error("Gcc is needed for the test to be run")
+        hackbench = self.fetch_asset('http://people.redhat.com'
+                                     '/~mingo/cfs-scheduler/'
+                                     'tools/hackbench.c')
+        shutil.copyfile(hackbench, os.path.join(self.srcdir, 'hackbench.c'))
+
+        os.chdir(self.srcdir)
+
+        if 'CC' in os.environ:
+            cc = '$CC'
+        else:
+            cc = 'cc'
+        process.system('%s  hackbench.c -o hackbench -lpthread' % cc)
+
+    def test(self):
+
+        time_value = self.params.get('time_val', default=None)
+        num_groups = self.params.get('num_groups', default=10)
```

Comment:
If you do the `self.params.get()` on `setUp()`, the parameters will be available for all tests. Ok, you have only one here, but still, it's a good practice.